apps/src/blockly/addons/cdoSpritePointer.ts

```diff
@@ -68,53 +75,72 @@ export function updatePointerBlockImage(
  *  should have a connection to a block of type gamelab_allSpritesWithAnimation, or we will return an empty string.
  * @returns Image url or empty string if the image source block does not have an image at the given index.
  */
-function getImageUrlFromImageSource(imageSourceBlock, imageIndexOnSource) {
+function getImageUrlFromImageSource(
+  imageSourceBlock: ExtendedBlockSvg,
+  imageIndexOnSource: number
+) {
   const targetConnection =
     imageSourceBlock.inputList[imageIndexOnSource]?.connection
       ?.targetConnection;
   // We only want to get the image from a connection block that is not an insertion marker.
   // If the block is an insertion marker that means the block is being dragged
   // and is not yet connected to the image source block.
+  const sourceBlock = targetConnection?.getSourceBlock();
   if (
-    targetConnection &&
-    targetConnection.sourceBlock_ &&
-    targetConnection.sourceBlock_.type === 'gamelab_allSpritesWithAnimation' &&
-    !targetConnection.sourceBlock_.isInsertionMarker()
+    sourceBlock &&
+    sourceBlock.type === 'gamelab_allSpritesWithAnimation' &&
+    !sourceBlock.isInsertionMarker()
   ) {
     // Blocks of type gamelab_allSpritesWithAnimation have an input with one field (the costume
     // picker dropdown).
-    return (
-      targetConnection.sourceBlock_.inputList[0].fieldRow[0].imageElement.getAttribute(
-        'xlink:href'
-      ) || ''
-    );
+    // imageElement is a private property on FieldDropdown.
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const imageElement = (sourceBlock.inputList[0].fieldRow[0] as any)
+      .imageElement as SVGImageElement;
+    return imageElement?.getAttribute('xlink:href') || '';
   } else {
     return '';
   }
 }
 
 // Set the given block's image input to the given imageUrl,
 // or reset the image input to the default long text if the imageUrl is empty.
-function changePointerImage(imageUrl, block) {
+function changePointerImage(
+  imageUrl: string | undefined,
+  block: ExtendedBlockSvg
+) {
   if (!imageUrl || imageUrl.length === 0) {
     resetPointerImageToLongString(block);
     return;
   }
-  const textInput = block.inputList[0].fieldRow[0];
```

Comment:
almost the exact same code was used here and in `resetPointerImageToLongString`. I pulled it into a helper method `updatePointerImageHelper`.